### PR TITLE
Device ID Refactor

### PR DESCRIPTION
**Pull request by rc14193**
Restructures device creation so that BLE is created from the Device ID and DeviceID class no longer carries the entire BLE class.

### CODE REVIEW

File: enlighten/Controller.py [hunk 1]
```diff
@@ -541,24 +543,15 @@ def connect_new(self, other_device=None):
...
+        self.bus.device_ids.remove(new_device_id)
+        if new_device_id in self.other_devices:
+            self.other_devices.remove(new_device_id)
 
         # attempt to connect the device
         if new_device_id.is_andor():
             self.marquee.info("connecting to XL spectrometer (please wait)", persist=True)
         else:
-            self.marquee.info("connecting to 0x%04x:0x%04x" % (new_device_id.vid, new_device_id.pid), persist=True)
+            self.marquee.info(f"connecting to {new_device_id}", persist=True)
```
Comment (rc14193) [inline enlighten/Controller.py:554]: I should add as it's easy to miss, the marquee will now say <DeviceID {BLE | USB | MOCK} {INFO}>. Not sure if you want maybe the DeviceID and <> stripped.

File: enlighten/BLEManager.py [hunk 2]
```diff
@@ -102,19 +106,18 @@ def ble_discover_click(self, btn, device):
 
     def perform_connect(self, btn, device):
         log.debug(f"called to perform connect on btn {btn} with text {btn.text()}")
-        self.ble_device = BLEDevice(device, self.loop)
-        ok = self.controller_connect(self.ble_device)
+        #self.ble_device = BLEDevice(device, self.loop)
+        self.ble_device_id = DeviceID(label=f"BLE:{device.address}:{device.name}")
```
Comment (mzieg) [inline enlighten/BLEManager.py:110]: I look forward to getting one of these units to test with :-)  (An actual BLE SiG, though I realize I could do with an RPi now...)

File: enlighten/Controller.py (hunk 1 above)
Comment (mzieg) [reply enlighten/Controller.py:554]: I'm okay with the verbose device description for now.  We can do a "friendliness / simplification" pass later, when all the cool new stuff is fully tested and working...it's reasonable to default to over-informative whilst in R&D.

File: scripts/Enlighten.py [hunk 3]
```diff
@@ -119,7 +119,7 @@ def run(self):
         # The sim_spec code is used for debugging with the test framework during peculiar issues
         #####
         self.controller.control_exit_signal.exit.connect(self.closeEvent)
-        #sim_spec = MockUSBDevice("SiG_785","EEPROM-EM-9c65d19f4c.json")
+        #sim_spec = DeviceID(label="MOCK:SiG_785:EEPROM-EM-9c65d19f4c.json")
```
Comment (mzieg) [inline scripts/Enlighten.py:122]: Thank you for maintaining even commented-out code :-)

**Review by mzieg — APPROVED** [review]
Awesome.

**Pull request merged**